schematic/__init__.py

```diff
@@ -159,11 +207,48 @@ def set_up_tracing(session: requests.Session) -> None:
         attribute_propagator = AttributePropagatingSpanProcessor(["user.id"])
         trace.get_tracer_provider().add_span_processor(attribute_propagator)
         exporter = OTLPSpanExporter(session=session)
+        # Overwrite the _readable_span method to redact sensitive data
+        SpanSdk._readable_span = _readable_span_alternate
         trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(exporter))
     else:
         trace.set_tracer_provider(TracerProvider(sampler=ALWAYS_OFF))
 
 
+original_function_readable_span = SpanSdk._readable_span
+
+
+def _readable_span_alternate(self: SpanSdk) -> ReadableSpan:
+    """Alternative function to the readable span. This function redacts sensitive data from the span attributes and events.
+
+    Args:
+        self (SpanSdk): _readable_span method of the SpanSdk class
+
+    Returns:
+        ReadableSpan: a new readable span that redacts sensitive data
+    """
+    redacted_events = []
+    # remove sensitive information in span status description
+    # this avoid statusMessage on signoz contains sensitive information
+    if self._status.status_code == trace.StatusCode.ERROR:
+        status_description_redacted = redact_string(str(self._status.description))
+        self._status = Status(self.status.status_code, status_description_redacted)
+
+    # remove sensitive information in attributes
+    # this avoid exception trace and messages contain sensitive information
+    for event in self._events:
+        attributes = event.attributes
+        redacted_event_attributes = redacted_sensitive_data_in_exception(attributes)
+        redacted_event = Event(
+            name=redact_string(event.name),
+            attributes=redacted_event_attributes,
+            timestamp=event.timestamp,
+        )
```

Comment:
Instead of creating and assigning a new `Event` is it possible to instead set the `name` attribute on `event` instead?

```suggestion
        event._name = redact_string(event.name)
```

The reason for this suggestion is that if there happens to be other fields that get added to the `Event`, this code is now incorrect since data would be dropped at this point in time. By modifying only the fields we want, then we have a lesser burden of maintenance and risk of breaking our code as the library is updated.

Reply:
Got it. Thanks for the suggestion. Let me give that a try.